test/specs/errors/tests/07.http_409_error.js

```diff
@@ -137,13 +146,26 @@ exports.execute = function (options) {
            });
        });
 
-        it("if it's a duplicate key error, we should generate a more readable message.", function (done) {
-            nock(url, ops)
-                .get("/ermrest/catalog/1235")
-                .reply(409, duplicateErrorServerResponse)
-                .persist();
+        it("on create, if it's a duplicate key error, we should generate a more readable message.", function (done) {
+            duplicateReference.contextualize.entryCreate.create([{duplicate_id: 1}]).then(null, function (err) {
+                expect(err.code).toBe(409, "invalid error code");
+                expect(err.message).toBe(duplicateErrorMappedMessage, "invalid error message");
```

Comment:
You should test the new attribute that you've added here.

Reply:
And since testing in ermrestjs is easier, it's better if you add a test case for the composite key in here too.